quinn/src/connection.rs

```diff
@@ -111,17 +107,19 @@ impl Future for ConnectionDriver {
             keep_going |= conn.handle_timer_updates();
             conn.forward_endpoint_events();
             conn.forward_app_events();
-            if !keep_going || conn.closed {
+            if !keep_going || conn.inner.is_drained() {
                 break;
             }
         }
 
         if !conn.inner.is_drained() {
-            Ok(Async::NotReady)
-        } else if let Some(e) = &conn.pending.error {
-            Err(e.clone())
-        } else {
-            Ok(Async::Ready(()))
+            conn.driver = Some(task::current());
```

Comment:
Is it necessary to do this even if `driver` is `Some` already? (I don't know how expensive it is.)

Reply:
This should cost at most an `Arc::clone`. I'm not certain off the top of my head whether futures are allowed to move between tasks; if they are, this is absolutely necessary.

Reply:
Fair enough, it does seem like the safer option.